internal/impl/postgresql/pglogicalstream/snapshotter.go

```
@@ -156,6 +159,69 @@ func (s *Snapshotter) prepareScannersAndGetters(columnTypes []*sql.ColumnType) (
 		case "INT4":
 			scanArgs[i] = new(sql.NullInt64)
 			valueGetters[i] = func(v interface{}) interface{} { return v.(*sql.NullInt64).Int64 }
+		case "JSONB":
+			scanArgs[i] = new(sql.NullString)
+			valueGetters[i] = func(v interface{}) interface{} {
+				payload := v.(*sql.NullString).String
+				if payload == "" {
+					return nil
+				}
+				var dst any
+				if err := json.Unmarshal([]byte(v.(*sql.NullString).String), &dst); err != nil {
+					s.logger.Warnf("Failed to unmarshal JSONB value: %v", err)
```

Let's return an error instead of logging